Important: #95647 - Composer installations and extension usage (#1585)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ExtensionArchitecture/DeclarationFile/Index.rst

.. _extension-declaration:

=========================================
Declaration file (:file:`ext_emconf.php`)
=========================================

> *-- required*  in legacy installations

Reviewer: ```suggestion
*-- required*  in Legacy installations
```

> The :file:`ext_emconf.php` is used in legacy non-Composer installations to

Reviewer: ```suggestion
The :file:`ext_emconf.php` is used in Legacy installations to
```

> supply information about the extension to the :guilabel:`Extension Manager`. In

Reviewer: ```suggestion
supply information about the extension in the :guilabel:`Extensions` module. In
```

There is no *Extension Manager* label in the TYPO3 backend, only the *Extensions* module.

these installations the ordering of installed extensions and their dependencies

> are loaded from this file as well.

Reviewer: ```suggestion
depends on this file as well.
```

.. versionchanged:: 11.4
   The ordering of installed extensions and their dependencies are loaded from
   the :file:`composer.json` file, instead of :file:`ext_emconf.php` in

>    composer-based installations.

Reviewer: ```suggestion
   Composer installations.
```

The only thing included

> is an associative array, :php:`$EM_CONF[extension key]`.

Reviewer: ```suggestion
is an associative array, :php:`$EM_CONF[$_EXTKEY]`.
```

The keys are described in the table below.

This file is overwritten, when extensions are imported from the online

> repository. So don't write your custom code in this file - only change

Reviewer: ```suggestion
repository (TER). So don't write your custom code in this file - only change
```

           Loading order especially matters when overriding TCA or SQL of another extension.

         The above example indicates that the extension depends on a
         version of TYPO3 between 10.4 and 11.5 (as only bug and security fixes are
         integrated into TYPO3 when the last digit of the version changes, it is
         safe to assume it will be compatible with any upcoming version of the
         corresponding branch, thus ``.99``). Also the extension has been
         tested and is known to work properly with PHP 7.4. and 8.0 It
         will conflict with "templavoilaplus" (any version) and it is suggested
         that it might be worth installing "news" (version at least 9.0.0).
         Be aware that you should add *at least* the TYPO3 and PHP version constraints
         to this file to make sure everything is working properly.

>          For legacy non-Composer installation the :file:`ext_emconf.php` file

Reviewer: ```suggestion
         For Legacy installations the :file:`ext_emconf.php` file
```
